### ISSUE TITLE
Overridden methods can't have different actor isolation from nonisolated overridden declaration

### ISSUE THREAD

**Issue by NachoSoto** [swiftlang/swift#75815]
### Description

This is the same as https://github.com/swiftlang/swift/issues/75732, but a different example from `init`.
This makes working with `XCTest` a pain now.

### Reproduction

```swift
@MainActor
final class MainActorType {
  init() {}
}

@MainActor
final class TestsForMainActorType: XCTestCase {
  // Main actor-isolated instance method 'setUp()' has different actor isolation from nonisolated overridden declaration
  @MainActor
  override func setUp() {
    super.setUp()

    // Removing `@MainActor` means this can't be called:
    // Call to main actor-isolated initializer 'init()' in a synchronous nonisolated context
    type = .init()
  }

  private var type: MainActorType?
}
```

### Expected behavior

This compiles

### Environment

```
swift-driver version: 1.113 Apple Swift version 6.0 (swiftlang-6.0.0.7.6 clang-1600.0.24.1)
Target: arm64-apple-macosx14.0
```

**Comment by NachoSoto** [comment]
cc @hborla

**Comment by NachoSoto** [comment]
This is just one particular example with `XCTest`, but the bigger issue is that there are still lots of Apple frameworks that aren't ready for `Swift 6`, like `Combine`.

**Comment by jonreid** [comment]
Basically anything that calls UIKit from any of the setUp/tearDown methods. This is a major impediment.

**Comment by s4cha** [comment]
Having the same issue on [Stevia](https://github.com/freshOS/Stevia) unit tests.
One trick is to make `setUp` `async throws` like so:
```
@MainActor class LayoutTests: XCTestCase {
    
    var v: UIView!
    
    override func setUp() async throws {
//        try await super.setUp() -> errors: Sending main actor-isolated value of type 'XCTestCase' with later accesses to nonisolated context risks causing data races
        v = UIView()
    }
```
But then calling super.setUp() errors out.
 @NachoSoto any news?

**Comment by NachoSoto** [comment]
Nope. The reality is that XCTest (like Combine and many other Apple frameworks) isn't ready for Swift Concurrency.

**Comment by s4cha** [comment]
Hey, I must have lived under a rock but I just realized new [Swift Testing](https://developer.apple.com/xcode/swift-testing) was I out so in my case it's just a matter of migrating XCTests to new Swift Testing \o/

**Comment by NachoSoto** [comment]
@hborla I think this is solved by [SE-0470](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0470-isolated-conformances.md)?

**Comment by hborla** [comment]
Yes that's right, SE-0470 is only for protocol conformances. I'm working on a proposal to apply similar techniques to subclasses to allow isolated overrides like this (using the basic approach outlined in the [vision document](https://github.com/swiftlang/swift-evolution/blob/main/visions/approachable-concurrency.md#isolated-subclasses-and-overrides)). I think there are also similar challenges with metatypes.
